Add task to export existing config in drupal.php

ramil-ubc opened the pull request:
This pull request introduces the initial implementation of the config backup feature.

ramil-ubc committed:
Add task to export existing config in drupal.php

ramil-ubc committed:
Refactor drush:config:export task between Drupal recipes

- Remove drush:config:export task from drupal.php as it exports configuration to a temporary location.
- Introduce enhanced drush:config:export task in drupal8.php, which now exports configuration to a backup directory within the specific release directory, improving the configuration management across different releases.

joelpittet commented on recipes/drupal8.php:
```diff
@@ -20,3 +20,11 @@
+  $destination = '{{deploy_path}}/releases/' . $prev_release . '/config/backup';
+  run("mkdir $destination");
+  drush("config:export --destination=$destination", ['runInCurrent', 'showOutput']);
+})->once();
```
You will want to remove the `once()` call so that this export happens on all servers (saves us from hunting down the live server at the time of the export)

joelpittet commented on recipes/drupal8.php:
```diff
@@ -20,3 +20,11 @@
+
+task('drush:config:export', function() {
+  $releasesList = get('releases_list');
+  $prev_release = max($releasesList);
```
This collecting `$prev_release` is not necessarily the previous or current release. Consider if you rolled back or if the next release is already happening during the deploy. There is probably a variable with info you want already in the code somewhere.

joelpittet commented on recipes/drupal8.php:
```diff
@@ -20,3 +20,11 @@
+  $releasesList = get('releases_list');
+  $prev_release = max($releasesList);
+  $destination = '{{deploy_path}}/releases/' . $prev_release . '/config/backup';
+  run("mkdir $destination");
```
It's worth it to add a `-p` flag to ensure all directories exist in the `mkdir` command. You will see Deployer doing this all over it's code.

joelpittet reviewed (changes requested):
It's looking pretty concise, here are some comments on what you have so far.

ramil-ubc commented:
@joelpittet When I tested this, the mkdir seems to be working but the drush config export wasn't. Do you have any idea why?

joelpittet commented:
@ramil-ubc I added some notes and questions to the AC task that might help with that, otherwise we can look together what you are seeing/not seeing in your test. 

The comment above about `mkdir` is not that it wouldn't work just a best practice/hardening of the code you have.

ramil-ubc committed:
Update configuration export task to handle backup more effectively

- Rename task from 'drush:config:export' to 'drush:config:backup' for clearer intent.
- Modify the backup destination path to prioritize the previous release directory, if available, otherwise use the current release path.
- Ensure creation of necessary directories with 'mkdir -p'.
- Change directory command to run within the release or current path before executing Drush.
- Enhance the Drush command for exporting configuration and add a confirmation message displaying the backup location.
- Set 'drush:config:backup' to run before 'deploy:drush' to ensure configurations are backed up prior to deployment tasks.

joelpittet commented on recipes/drupal8.php:
```diff
@@ -20,3 +20,18 @@
 })->once();
+
+task('drush:config:backup', function() {
+  if (has('previous_release')) {
```
It would be good to bail out/skip when the .env file is missing.

joelpittet commented:
Merging this in to export any production config diffs. We can make improvements to how we use the export tomorrow when patching.
